WonderMake/Utilities/Container/Container.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+#include <concepts>
+#include <type_traits>
+
+#include <vector>
+#include <list>
+
+#include "Utilities/plf_colony.h"
+
+#include "Traits/ContainerTraits.h"
+#include "BackendImplementations/ColonyBackend.h"
+#include "BackendImplementations/LinkedListBackend.h"
+#include "BackendImplementations/VectorBackend.h"
+
+#include "Utilities/TypeTraits/ParameterPack.h"
+#include "Utilities/Container/Traits/Implications.h"
+#include "Utilities/TypeTraits/ContainsType.h"
+#include <iostream>
+
+template<typename TObjectType, typename TContainerBackends, ContainerTrait ... TContainerTraits>
+class ContainerBase
+{
+	//[Nicos]: These need to be at the top so we can use Storage as a type
+private:
+	using TraitsPack = typename ResolvedImplications<TContainerTraits...>::type;
+
+	template <typename ... TBackends>
+	constexpr static auto StorageType(ParameterPack<TBackends...>)
+	{
+		return FirstSatisfyingBackend<TraitsPack, TBackends...>::type();
+	}
+
+	using Storage = decltype(StorageType(TContainerBackends()));
+
+	template <typename T>
+	constexpr static bool UsesStorage = std::is_same_v<Storage, T>;
+
+	Storage myBackend;
+
+public:
+
+	void PrintTraits()
+	{
+		std::cout << typeid(TraitsPack).name() << std::endl;
+	}
+
+	template <typename T>
+	constexpr static bool HasTrait = IsSubsetOf<ParameterPack<T>, TraitsPack>::value;
+	constexpr static bool Iterable = HasTrait<Iterable>;
+	constexpr static bool Indexable = HasTrait<Indexable> || HasTrait<RandomAccess>;
+	constexpr static bool RandomAccess = HasTrait<RandomAccess>;
+	constexpr static bool StableElements = HasTrait<StableElements>;
+	constexpr static bool ConstantInsertion = HasTrait<ConstantInsertion>;
+	constexpr static bool ConstantDeletion = HasTrait<ConstantDeletion>;
+	constexpr static bool Sortable = HasTrait<Sortable>;
+	constexpr static bool CanEqualityCompare = std::equality_comparable<TObjectType>;
+
+	template<typename TBackend>
+	constexpr bool HasBackend() const 
+	{
+		return typeid(TBackend) == typeid(Storage);
+	}
+
+	template<typename TObjectTypeFunc> requires std::is_same_v<TObjectType, std::decay_t<TObjectTypeFunc>> || std::is_constructible_v<TObjectType, TObjectTypeFunc>
+	void Add(TObjectTypeFunc aObject)
+	{
+		myBackend.Add<TObjectTypeFunc>(std::forward<TObjectTypeFunc>(aObject));
+	}
+
+	auto Find(const TObjectType& aObject) requires Iterable && CanEqualityCompare
+	{
+		auto it = std::find(begin(), end(), aObject);
+		return it;
```

Comment:
return std::find(begin(), end(), aObject);

Reply:
It be removed